### PR TITLE
Drop Python 3.5

### LINKED ISSUE #543: Drop Python 3.5
and update black.

See https://github.com/marshmallow-code/apispec/pull/582

### PR DESCRIPTION
**Pull request by sirosen**
Also update pre-commit config and relevant docs.

There are also a bunch of changes from pyupgrade.

closes #543

### CODE REVIEW

File: src/webargs/asyncparser.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-"""Asynchronous request parser. Compatible with Python>=3.5."""
+"""Asynchronous request parser. Compatible with Python>=3.6."""
```
Comment (lafrech) [inline src/webargs/asyncparser.py]: This Python version compat comment should be removed.

**Review by lafrech — COMMENTED** [review]
Thanks for this !

TODO

- Remove comment in async parser
- Update tox.ini: remove py35 (and perhaps update marshmallow>=3.0.0rc2, unrelated but while we're at it)

**Comment by sirosen** [comment]
I knew I must have missed something! 😄 
I'm guessing that comment was added for python2 users, but I wasn't sure.

I've pushed a commit with the changes.

**Comment by lafrech** [comment]
Good. Thanks.

Yes, that comment dates back to when Python2 was still supported and the async parser was in a dedicated py3 directory.

Approved. :+1:

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,15 +1,15 @@
 repos:
 - repo: https://github.com/asottile/pyupgrade
-  rev: v2.7.2
+  rev: v2.7.3
   hooks:
   - id: pyupgrade
-    args: ["--py3-plus", "--keep-mock"]
+    args: ["--py36-plus", "--keep-mock"]
```
Comment (sloria) [inline .pre-commit-config.yaml:6]: Do we still need `--keep-mock`?

Comment (sirosen) [reply .pre-commit-config.yaml:6]: Good catch! I don't think it does anything now that we're not using the backport. I waited to reply until I got to confirm this.
I just opened #556 to clean up.

**Pull request merged**